hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/dog/RoutineEnvironmentTransformations.scala

```diff
@@ -0,0 +1,370 @@
+    // ONLY if: de_sleep_day_yn = 1
+    val daytimeSleepLocation =
+      if (daySleep.contains(true)) rawRecord.getOptionalNumber("de_sleep_location_day") else None
+    // ONLY if: de_sleep_location_day = 98
+    val daytimeSleepLocationOtherDescription =
+      if (daytimeSleepLocation.contains(98L)) rawRecord.getOptional("de_sleep_loc_day_other")
+      else None
+    dog.copy(
+      deNighttimeSleepLocation = nighttimeSleepLocation,
+      // ONLY if: de_sleep_location = '98'
+      deNighttimeSleepLocationOtherDescription =
+        if (nighttimeSleepLocation.contains(98L)) rawRecord.getOptional("de_sleep_location_other")
+        else None,
+      deNighttimeSleepAvgHours = rawRecord.getOptionalNumber("de_sleep_amt_night"),
+      deDaytimeSleepLocationDifferent = daySleep,
+      deDaytimeSleepLocation = daytimeSleepLocation,
+      deDaytimeSleepLocationOtherDescription = daytimeSleepLocationOtherDescription,
+      deDaytimeSleepAvgHours = rawRecord.getOptionalNumber("de_sleep_amt_day")
+    )
+  }
+
+  /* Toxins ingested */
+  def mapToxinsIngested(
+    rawRecord: RawRecord,
+    dog: HlesDogRoutineEnvironment
+  ): HlesDogRoutineEnvironment = {
+    val toxinsAmount = rawRecord.getOptionalNumber("de_ingest_bad_amt")
+    //ONLY if: de_ingest_bad_amt = 1 OR de_ingest_bad_amt = 2
+    if (toxinsAmount.exists(_ > 0)) {
+      val toxinsDescription = rawRecord.getOptional("de_ingest_bad")
+      val toxinsIngested = rawRecord.get("de_ingest_bad_what")
+      val recentToxinsOrHazardsIngestedOther = toxinsIngested.map(_.contains("98"))
+      dog.copy(
+        deRecentToxinsOrHazardsIngestedFrequency = toxinsAmount,
+        deRecentToxinsOrHazardsIngestedChocolate = toxinsIngested.map(_.contains("1")),
+        deRecentToxinsOrHazardsIngestedPoison = toxinsIngested.map(_.contains("2")),
+        deRecentToxinsOrHazardsIngestedHumanMedication = toxinsIngested.map(_.contains("3")),
+        deRecentToxinsOrHazardsIngestedPetMedication = toxinsIngested.map(_.contains("4")),
+        deRecentToxinsOrHazardsIngestedGarbageOrFood = toxinsIngested.map(_.contains("5")),
+        deRecentToxinsOrHazardsIngestedDeadAnimal = toxinsIngested.map(_.contains("6")),
+        deRecentToxinsOrHazardsIngestedToys = toxinsIngested.map(_.contains("7")),
+        deRecentToxinsOrHazardsIngestedClothing = toxinsIngested.map(_.contains("8")),
+        deRecentToxinsOrHazardsIngestedOther = recentToxinsOrHazardsIngestedOther,
+        //de_ingest_bad is deprecated, check it first, then check de_ingest_bad_what_other
+        deRecentToxinsOrHazardsIngestedOtherDescription =
+          if (toxinsDescription.exists(_.trim.nonEmpty)) toxinsDescription
+          // ONLY if: de_ingest_bad_what(98) = 1
+          else if (recentToxinsOrHazardsIngestedOther.get)
+            rawRecord.getOptional("de_ingest_bad_what_other")
+          else None,
+        deRecentToxinsOrHazardsIngestedRequiredVet =
+          rawRecord.getOptionalBoolean("de_ingest_bad_er_yn")
```

Comment:
Any suggestions on cleaning up/refactoring this to be more readable?
This particular piece was pretty annoying.

Reply:
I think something like this would work:
```scala
recentToxinsOrHazardsIngestedOther.fold(toxinsDescription) {
  if (_) rawRecord.getOptional("de_ingest_bad_what_other") else None
}
```
And I'd tweak the definition of `toxinsDescription` to include the trim & filter:
```scala
val toxinsDescription = rawRecord.getOptional("de_ingest_bad").map(_.trim).filter(_.nonEmpty)
```